packages/ragbits-core/src/ragbits/core/embeddings/vertex_multimodal.py

```diff
@@ -0,0 +1,163 @@
+import asyncio
+import base64
+
+try:
+    import litellm
+    from litellm.llms.vertex_ai_and_google_ai_studio.common_utils import VertexAIError
+    from litellm.main import VertexMultimodalEmbedding
+
+    HAS_LITELLM = True
+except ImportError:
+    HAS_LITELLM = False
+
+from ragbits.core.embeddings import Embeddings
+from ragbits.core.embeddings.exceptions import (
+    EmbeddingResponseError,
+    EmbeddingStatusError,
+)
+
+
+class VertexAIMultimodelEmbeddings(Embeddings):
```

Comment:
I don't get why we have a separate class for this. We are using litellm under the hood here, it'd be more intuitive to implement `embed_image` in `LiteLLMEmbeddings` and constrain it to VertexAI embeddings.

Reply:
That's the problem I tried to described during the daily today. In theory we still use LiteLLM but with this model LiteLLM behaves differently in almost every way you could imagine:

1. It expects a different input format
2. It returns output in a different format
3. It throws different exceptions when errors occur
4. It doesn't support embedding multiple things at once (causing me to emulate it with async, looping and semaphores in `VertexAIMultimodelEmbeddings`) 

I have a working implementation of `LiteLLMEmbeddings` with VertexAI Multimodal Embeddings support that has workarounds around points 1 and 2. Issue 4 caused me to abandon this direction altogether. Not only having both implementations (the one that can send multiple items to the model at once, and the one that emulates this with async, looping and semaphores) in a  single class would bring a lot of complexity to it, but even more importantly I don't have a condition that would allow me to switch between those two behaviors in a way that makes sense in a general implementation (i.e., other than switching based on the exact name of a single, niche model). That's why, for now, I decided to keep this separate from `LiteLLMEmbeddings`, in order to keep its implementation general and sane. I think we should revisit this in the future, but in my mind having a support for images in `LiteLLMEmbeddings` would require helping LiteLLM deal at least with some of the issues listed above - i.e., helping it provide an unified interface that we could use.

Reply:
Ok, makes sense to me